OpenSim/Common/Test/CMakeLists.txt

```diff
@@ -15,5 +15,5 @@ OpenSimAddTests(
     TESTPROGRAMS ${TEST_PROGS}
     DATAFILES ${TEST_FILES} ${C3D_TEST_FILES} ${TRC_TEST_FILES} 
               ${MOT_TEST_FILES}
-    LINKLIBS ${BTK_LIBRARIES} osimCommon ${SIMTK_ALL_LIBS} 
+    LINKLIBS ${BTK_LIBRARIES} osimCommon osimSimulation ${SIMTK_ALL_LIBS} 
```

Comment:
There are currently tests for `ConsoleReporter` and `TableReporter`, but they are of the "does it run?" kind, not of the "does it run correctly?" kind. 

The new `testReporters` is certainly valuable. A middle ground would be to view the proposed `testReporters` as a test of the functioning of Reporters with a proper Model. So we could rename that test to `testReportersWithModel` and leave it in Simulation.